File: 950. Reveal Cards In Increasing Order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck.sort()
        q = collections.deque()
        for i in range (len(deck)-1, -1, -1):
            if q:
                q.appendleft(q.pop())
            q.appendleft(deck[i])
        return list(q)

File: 950. Reveal Cards In Increasing Order.go
func deckRevealedIncreasing(deck []int) []int {
	sort.Ints(deck)
	q := []int{}
	for i := len(deck) - 1; i >= 0; i-- {
		if len(q) > 0 {
			q = append([]int{q[len(q)-1]}, q[:len(q)-1]...)
		}
		q = append([]int{deck[i]}, q...)
	}
	return q
}